### PR TITLE
Profile edit picture UI

### LINKED ISSUE #11280: Allow user to select a profile picture from gallery
https://github.com/status-im/status/issues/4

As a user 
I want to be able to select a profile picture from gallery
So that it can be seen by other status users when chatting with them

### Description

The user should be able to select a picture from gallery (which format do we want to support), which will be selected as their profile picture.
This picture should be shown in chat/profile/all the relevant places.

https://www.figma.com/file/TNCyHKtR3sx5EL6YznFWUa4O/Profile?node-id=510%3A3116

### Notes

There used to be some code already for this, and you might be able to reuse it 

This can be worked together with #11281 #11282

### LINKED ISSUE #11281: Allow user to set profile picture from camera
https://github.com/status-im/status/issues/4

As a user
I want to be able to take a picture from camera
So that it can be seen by other status users when chatting with them

Description
The user should be able to take a picture from camera (which format do we want to support), which will be selected as their profile picture.
This picture should be shown in chat/profile/all the relevant places.

https://www.figma.com/file/TNCyHKtR3sx5EL6YznFWUa4O/Profile?node-id=510%3A3116

Notes
There used to be some code already for this, and you might be able to reuse it

This can be worked together with #11280 #11282

### LINKED ISSUE #11282: Allow user to crop profile picture
As a user
I want to be able to crop a profile picture
So that it can be nicely shown to other users

Description
The user should be able to crop an image as per designs, which will be selected as their profile picture.
This picture should be shown in chat/profile/all the relevant places.

https://www.figma.com/file/TNCyHKtR3sx5EL6YznFWUa4O/Profile?node-id=510%3A3116

### PR DESCRIPTION
**Pull request by Ferossgp**
Fixes #11281 Fixes #11280 Fixes #11282

### CODE REVIEW

**Comment by cammellos** [comment]
@churik this should be ready to be tested, being quite a large PR we can review it in the meantime if that's ok, as likely some issuse will crop up.

How it should work:

1) If you set your profile picture, it will be shown to people that are in your public chat only if you posted in that public chat after setting your profile picture.
2) If I have added you as a contact or opened a one to one chat with you, I should see your profile picture if set. 

2 has currently a known issue that is taking some time to update, looking into it.

Thanks!

**Comment by churik** [comment]
### ISSUE 1: empty picture instead of default in case of removing profile picture
**Steps:**
1) restore multiaccount from seed phrase
2) set profile picture
3) remove profile picture

**Expected result:**
autogenerated user pic
**Actual result:**
IMAGE 2020-12-10 15:25:37

**OS:** IOS, Android

### ISSUE 2: default profile picture instead of custom in user status

**Steps:**
1) User A: restore multiaccount from seed phrase
2) User A: set profile picture
3) User A: post smth in timeline (Status > + > Add message)
4) User B: open 

**Expected result:**
custom userpicture 

**Actual result:**
Monosnap 2020-12-10 16-07-52

**OS:** IOS, Android

### ISSUE 3: no profile pictures in mention list

Profile pictures of other users are not shown in mentions list, my profile is shown with blank profile image
Monosnap 2020-12-10 16-09-33

### ISSUE 4: if User A was added as contact after he added User B as contact, User B will not see custom picture of User A in 1-1 chat

**Steps:**
1) User A: start chat with User B, send several messages
2) User B: send several messages to User A
3) User A: set profile picture
4) User B: add User A as contact
5) User A: send several messages to user B. (1-1 chat on User B device will be updated with custom userpic of User A)
6) User B: set profile picture
7) User A: add User B as contact
6) User B:  send several messages to User A

**Expected result:**
1-1 chat on User B device will be updated with custom userpic of User B

**Actual result:**
default pic (will be updated for user B when he will post smth in public chat)

**OS:** IOS, Android

**Comment by Ferossgp** [comment]
@churik first 3 issues should be fixed, could not replicate 4, but it takes some time to update the picture depending on when the message with the profile picture will be handled

**Comment by cammellos** [comment]
4 Is a known issue, @Samyoul is fixing it, but we might fix it in a separate PR if that's ok

**Comment by churik** [comment]
@cammellos thank you!
Tested on builds with https://github.com/status-im/status-react/commit/d7d297b00e8c2d8b17f15ad5c065ed3d870fbdb8.
### Regarding ISSUE 1: empty picture instead of default in case of removing profile picture
If you remove profile picture and relogin:
1.1 First letter of your default username is shown in profile
1.2 If you post smth in timeline, your avatar is shown as empty
Monosnap 2020-12-11 10-23-29 png 2020-12-11 10-24-45
After creating new account and relogin empty picture in profile is shown.

IMAGE 2020-12-11 10:33:24

### Regarding ISSUE 4: profile picture of another user will be updated only in case if he will post smth in public chat, where both users are participants (regardless they are contacts or not)
Or time to wait for updates too long (>10 mins) 

### ISSUE 5: profile picture is not syncing across devices

**Steps:**
1) Device 1: restore multiaccount A from seed phrase, set profile picture
2) Device 2: restore multiaccount A from seed phrase
3) Pair both devices
4) Sync both devices

**Expected result:**
profile picture is updated on device 2
**Actual result:**
profile picture remains the same 
**OS:** IOS, Android

### ISSUE 6: no default image on "Your keys have been successfully recovered" screen
During restoring account from seed phrase:
IMAGE 2020-12-11 12:11:40


### ISSUE \ QUESTION 7: Editing(cropping, rotating etc) profile picture interface looks a bit odd on Android
Not sure that this is expected, also not sure if it can be changed at all
**Device:** Xiaomi Mi 9T Pro
Monosnap 2020-12-11 12-14-26

**Comment by cammellos** [comment]
ISSUE 5 if that's ok we'd also fix in a separate PR, as currently is not a supported functionality, if that's ok with you
thanks!

**Comment by Ferossgp** [comment]
ISSUE 7 is from the gallery, I could not find how to update it. 

@churik can you please also test the keycard flow? As there were also mentions of photos in the code.

**Comment by churik** [comment]
@Ferossgp 
about keycard:I retested relogin, creating account, rocovering with keycard, restoring multiaccount from seed phrase on keycard storage.
No issues found.

Please ping me when ready to test again.

**Comment by Ferossgp** [comment]
It is ready

**Comment by churik** [comment]
@Ferossgp can't get woking diawi build, details in [ci](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-11347/34/).
Can you please take a look?

**Comment by cammellos** [comment]
@churik I have pushed and resolved the conflicts.

**Comment by churik** [comment]
@cammellos 
the profile picture is not updated for another user at all on build with https://github.com/status-im/status-react/commit/923c97acba137e5160e5de35b0645c6bc6f193be 

**Steps:**
1) create multiaccounts A and B on different devices
2) A, B: post smth in public chat  
3) A adds B as contact, B adds A as contact
3) A updates profile picture
4) A posts smth in public chat

**Expected result:**
B will see updated profile picture of A

**Actual result:**
profile picture is not updated (tried post new messages in status timeline, 1-1 chat, tried to religion on both devices - profile picture will not update on device B for user A)

**OS:** IOS, Android

**Comment by cammellos** [comment]
@churik should be ready to test again, thanks for your patience.

**Comment by churik** [comment]
@cammellos testing is in progress.

### ISSUE 8: in mention list in 1-1 chat available all users with picture
**Steps:**
1) restore multiaccount from seed phrase
2) join to public chat where users have pictures (#chu) 
3) start 1-1 chat with some user
4) type @ in chat input field

**Expected result:**
only 1-1 chat member + user with ENS names are available

**Actual result:**
available all users with profile picture
IMAGE 2020-12-16 11:36:33

**OS:** IOS, Android

### ISSUE 9: after removing profile picture remains the same for other users

**Steps:**
1) User A: restore multiaccount from seed phrase
2) User A:  set profile picture
3) User A: send smth to public chat where User B is participated
4) User A: remove profile picture
5) User A: send smth in public chat
6) User B: check public chat

**Expected result:**
User B sees default profile picture of user A
**Actual result:**
profile picture is not changed

**OS:** IOS, Android

File: src/status_im/multiaccounts/core.cljs
Comment (Ferossgp) [inline src/status_im/multiaccounts/core.cljs:177]: Should be updated to the new format

Comment (cammellos) [reply src/status_im/multiaccounts/core.cljs:177]: it's still an array for multiaccount, I haven't changed it, it's a map for contacts (I don't want to make too many changes for now).
Do you think is still an issue?

Comment (Ferossgp) [reply src/status_im/multiaccounts/core.cljs:177]: No, I was thinking that format for all places was changed

**Comment by churik** [comment]
@cammellos found one more creepy issue: after cropping photo and setting it as profile picture sometimes app is just freezing (some components are missing, like "Send message" button, you can't log out as log out button is unresponsive etc)

Reproduced around six times (And Diana 5-6 times), in ~2-5% cases (so not really often) - but unfortunately still didn't find exact steps.
Basically what we did:
- select imsge from gallery
- resizing it
- go to chat view and send some message in public chat
On gif consequences of that issue:
BAACAgQAAxkBAAEuTvdf2gMhI3C5FuF-ziC7WAAB8COjQb0AAokIAAKqXtBSZMTJsUCqnZ0eBA_1608123207 73

Issue is reproducible only on Android.
Several logs:
[Status-debug-logs.zip](https://github.com/status-im/status-react/files/5702756/Status-debug-logs.zip)
[Status-debug-logs (6).zip](https://github.com/status-im/status-react/files/5702760/Status-debug-logs.6.zip)
DEBUG level: 
[Status-debug-logs.zip](https://github.com/status-im/status-react/files/5703239/Status-debug-logs.zip)

In all logs before this behavior was:
`lvl=info msg=resizing size=80 width=80 height=80 `


**UPD:**
Found steps.
It happens when changing profile picture with photos with different side ratio, i.e. in landscape and portrait mode.
Steps:
1) set profile picture in portrait mode, try to send message to public chat A and public chat B
2) set profile picture in landscape mode, try to send message to public chat A and public chat B
3) set profile picture in portrait mode, try to send message to public chat A and public chat B (it may require to repeat previous steps one more time)

After this app is freezing.

**Comment by cammellos** [comment]
@churik it should be ready to be tested again, we found the issue and I wasn't able to replicate after the fix.
Thanks!

**Comment by churik** [comment]
found one small issue, but thick can be fixed separately :)
send picture in chat -> tap on it -> options -> default userpicture is shown (IOS only, on Android  there is no option to view user profile)
IMAGE 2020-12-17 11:49:07

Waiting for e2e results, after this well be ready to merge.

**Review by cammellos — APPROVED** [review]

**Comment by churik** [comment]
Issues #11559 , #11558, #11557 are opened to fix separately.

No more issues for this PR.
e2e failures are not related to PR.
Tested basic flows, LGTM!

**Devices:**
- IPhone 12 Mini (IOS 14)
- IPhone 7 (IOS 13)
- Xiaomi Mi Note 9 Pro (Android 10)

**Pull request merged**
